File: 1º Ano/2º Semestre/Programação Orientada a Objetos/Labs/Lab06/CardGames/src/CardGames/FaceCard.java
package CardGames;

/**
 * Representa uma carta de jogar com uma figura.
 * 
 * @author POO 2019/2020
 * @version maio/2020
 */

public class FaceCard extends Card {
    private FaceName faceName;

    public FaceCard(FaceName faceName, Suit suit, int value) {
        super(suit, value);
        this.faceName = faceName;
    }

    public FaceCard(FaceName face, Suit suit) {
        super(suit);
        this.faceName = face;
    }

    public FaceName getFaceName() {
        return faceName;
    }

    @Override
    public String getName() {
        return faceName.toString(); 
    }

    @Override
    public String toString() {
        return faceName + " de " + getSuit();
    }
}

File: 3º Ano/Anual/Programação Visual (PV)/Labs/Lab 1/Lab1/FaceCard.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Lab1
{
    class FaceCard : Card
    {
        private FaceName faceName;

        public FaceCard(FaceName faceName, Suit suit, int value) : base(suit, value) => this.faceName = faceName;

        public FaceCard(FaceName faceName, Suit suit) : base(suit) => this.faceName = faceName;

        public FaceCard() : base(Suit.None) => this.faceName = FaceName.None;
        
        public override string Name => this.faceName.ToString();
        public FaceName FaceName => this.faceName;

        public override string ToString() => "Carta -> " + this.faceName.ToNameString() + " de " + base.GetSuit().ToNameString() + "\n";

    }
}
